news.js

```diff
@@ -0,0 +1,250 @@
+	},
+	{
+		image: "<div class='content__img'></div>",
+		heading: 'Fox News',
+		date: 'Posted on 29 June, 2019 //Category: Category One',
+		content: 'Djokovic reveals ‘silly’ mental trick to help him deal with crowd at Wimbledon',
+		popup: `Djokovic revealed a mental trick, according to the Times. After winning the match, after nearly five hours, he told reporters that when the crowd chanted “Roger! Roger,” he said he heard “Novak.”“It sounds silly, but it is like that. I try to convince myself that it’s like that.” The sets were 7-6 (5), 1-6, 7-6 (4), 4-6 and 13-12 (3). The final lasted nearly five hours, longer than any other Wimbledon final. Djokovic has now won Wimbledon five times.`
+	},
+	{
+		image: "<div class='content__img'></div>",
+		heading: 'CNN',
+		date: 'Posted on 29 June, 2019 //Category: Category One',
+		content: "'Silverstone, I love you': Lewis Hamilton wins record sixth British Grand Prix",
+		popup: `Lewis Hamilton claimed a thrilling victory at Silverstone on Sunday to win a record-breaking sixth British Grand Prix. It looked as though teammate Valtteri Bottas would spoil the party for the 141,000 visiting fans, but a timely safety car intervention gifted Hamilton first place. Ferrari's Charles Leclerc claimed an impressive third place after an intense, wheel-to-wheel battle with Max Verstappen throughout the race.`
+	}
+];
+//Channels present in the dropdown list
+let channels = ['BBC', 'Aaj Tak', 'NBC', 'NDTV', 'Republic', 'TV9', 'India Today', 'Times Now', 'Fox News', 'CNN'];
+//variable used to store the complete data of all news channels
+let fullData = ' ';
+//variable used to store the popup content
+let modalData = `<div id="myModal" class="modal">
+					<div class="modal-content">
+						<div class="modal-header">
+							<span class="close" onclick="closepopup()">&times;</span>
+							<h2 id="popup_head"></h2>
+						</div>
+						<div class="modal-body">
+							<p id="popup_content"></p>
+						</div>
+						<div class="modal-footer">
+							<h3 id="popup_foot">&copy; NewsFeed 2019</h3>
+						</div>
+					</div>
+				</div>`;
+
+//Function to display the selected news category
+function myFunction() {
+	let selectedCategory = document.getElementById('sel-category').value;
+	let display;
+	let indexDisplay;
+	if (selectedCategory !== 'All') {
+		let singleData = data.filter(function (e, index) {
+			if (selectedCategory === e.heading)
+				indexDisplay = index;
+			return selectedCategory === e.heading;
+		});
+		displaySingleNews(singleData, indexDisplay);
+	}
+	else if (selectedCategory === 'All') {
+		document.getElementById('displaynews').innerHTML = fullData;
+	}
+
+}
+function displaySingleNews(singleData, indexDisplay) {
```

Comment:
We dont need another method for single news. reuse the displayNews method only.

do like this.

displayNews(fullData)
displayNews(data)